sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzurePath.java

```diff
@@ -3,30 +3,81 @@
 
 package com.azure.storage.blob.nio;
 
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.nio.implementation.util.Utility;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileStore;
 import java.nio.file.FileSystem;
+import java.nio.file.InvalidPathException;
 import java.nio.file.LinkOption;
 import java.nio.file.Path;
 import java.nio.file.WatchEvent;
 import java.nio.file.WatchKey;
 import java.nio.file.WatchService;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
 import java.util.Iterator;
+import java.util.Objects;
 
 /**
+ * The root component, if it is present, is the first element of the path and is denoted by a {@code ':'} as the last
+ * character. Hence, only one instance of {@code ':'} may appear in a path string and it may only be the last character
+ * of the first element in the path. The root component is used to identify which container a path belongs to.
+ * <p>
+ * Constructing a syntactically valid path does not ensure a resource exists at the given path. An error will
```

Comment:
Could this be made into more of a bulleted list, kind of like

An error will only be thrown for 
- an invalid root when trying to access a file in that root if it does not exist
- a nonexistent path when trying to access the resource at that location if it does not exist or is inaccessible